Author: stevehay
Date: Tue Jun 27 08:38:15 2023
New Revision: 1910623

Stop using do_open9() - this deprecated function was removed in 5.37.1

Patch by Jitka Plesnikova <jp...@redhat.com> from https://rt.cpan.org/Ticket/Display.html?id=148451

Index: src/modules/perl/modperl_io.c
===================================================================
--- src/modules/perl/modperl_io.c	(revision 1910622)
+++ src/modules/perl/modperl_io.c	(revision 1910623)
@@ -116,8 +116,8 @@
     save_gp(handle, 1);
 
     sv_setref_pv(sv, "Apache2::RequestRec", (void*)r);
-    status = do_open9(handle, mode == O_RDONLY ? "<:Apache2" : ">:Apache2",
-                      9, FALSE, mode, 0, (PerlIO *)NULL, sv, 1);
+    status = do_openn(handle, mode == O_RDONLY ? "<:Apache2" : ">:Apache2",
+                      9, FALSE, mode, 0, (PerlIO *)NULL, &sv, 1);
     if (status == 0) {
         Perl_croak(aTHX_ "Failed to open STD%s: %" SVf,
                    mode == O_RDONLY ? "IN" : "OUT", get_sv("!", TRUE));
